pep-0613.rst

```diff
@@ -0,0 +1,173 @@
+
+
+Error Messaging:
+****************
+
+::
+
+  MyType = InvalidType
+  MyType = MyGeneric(int)  # i.e., intention was MyGeneric[int]
+
+A type checker should warn on this code snippet that :code:`InvalidType` is not a valid type, and therefore cannot be used to annotate an expression or to construct a type alias. Instead, type checkers are forced to throw spurious errors that (1) :code:`MyType` is a global expression missing an annotation, and (2) :code:`MyType` is not a valid type in all usages of :code:`MyType` across the codebase.
+
+::
+
+  MyType: TypeAlias = InvalidType
+  MyType: TypeAlias = MyGeneric(int)
+
+With explicit aliases, the type checker has enough information to error on the actual definition of the bad type alias, and explain why: that :code:`MyGeneric(int)` and `InvalidType` are not valid types. When the value expression is no longer evaluated as a global value, unactionable type errors on all usages of :code:`MyType` across the codebase can be suppressed.
+
+Scope Restrictions:
+*******************
+
+::
+
+  x = ClassName
+  def foo() -> None:
+    x = ClassName
+
+The outer :code:`x` is a valid type alias, but type checkers must error if the inner :code:`x` is ever used as a type. This is confusing because the alias declaration rule is not explicit, and the error is not thrown on the location of the inner type alias declaration but rather on every one of its subsequent use cases.
+
+::
+
+  x: TypeAlias = ClassName
+  def foo() -> None:
+    x = ClassName
+    x: TypeAlias = ClassName
+
+With explicit aliases, the outer assignment is still a valid type variable, and the inner assignment can either be a valid local variable or a clear error, communicating to the author that type aliases cannot be defined inside a nested scope.
+
+
+Specification
+=============
+
+The explicit alias declaration syntax attempts to clearly differentiate between the three possible kinds of assignments: typed global expressions, untyped global expressions, and type aliases. This avoids the existence of assignments that break type checking when an annotation is added, and avoids classifying the nature of the assignment based on the type of the value.
+
+Implicit syntax (pre-existing):
+
+::
+
+  x = 1  # untyped global expression
+  x: int = 1  # typed global expression
+
+  x = int  # type alias
+  x: Type[int] = int  # typed global expression
+
+
+Explicit syntax:
+
+::
+
+  x = 1  # untyped global expression
+  x: int = 1  # typed global expression
+
+  x = int  # untyped global expression (see note below)
+  x: Type[int] = int  # typed global expression
+
+  x: TypeAlias = int  # type alias
+  x: TypeAlias = “MyClass”  # type alias
+
+
+Note: The examples above illustrate implicit and explicit alias declarations in isolation. For the sake of backwards compatibility, type checkers may choose to support both simultaneously, meaning an untyped global expression :code:`x = int` will still be considered a valid type alias.
```

Comment:
I'm not happy with this. There must by now exist millions of lines of code using implicit aliases unambiguously. I would rather not declare those invalid at the whim of the type checker -- it would not promote interoperability. My preference is for `x = int` to be a valid type alias except in some corner cases (like `x = "MyClass"`), basically as things are today.

Reply:
I wrote this with the thought in mind that having an explicit-only system might be more straightforward than a mixed one and the conversion could be an easy codemod, so wanted to keep that option open. However, valuing interoperability makes sense - I'll adjust the wording here.